tools/tpm2_ptool.py

```diff
@@ -1063,175 +1291,90 @@ def generate_options(self, group_parser):
-                    encobjauth = binascii.hexlify(encobjauth)
-                else:
-                    c = AESCipher(binascii.unhexlify(wrappingauth))
-                    encobjauth = c.encrypt(objauth)
-
-                #
-                # Cache the objects attributes from the public structure and other sources
-                # and populate the db with the data. This allows use of the public data
-                # without needed to load any objects which requires a pin to do.
-                #
-                y = yaml.load(tertiarypubdata)
-
-                if alg.startswith('rsa'):
-                    attrs = [
-                        {  CKA_KEY_TYPE        : CKK_RSA         },
-                        {  CKA_CLASS           : CKO_PRIVATE_KEY },
-                        {  CKA_CLASS           : CKO_PUBLIC_KEY  },
-                        {  CKA_ID              : id              },
-                        {  CKA_MODULUS         : y['rsa']        },
-                        {  CKA_PUBLIC_EXPONENT : 65537           },
-                    ]
-
-                    mech = [
-                        { CKM_RSA_PKCS_OAEP : "" },
-                    ]
-
-                elif alg.startswith('aes'):
-                    attrs = [
-                        { CKA_CLASS    : CKO_SECRET_KEY },
-                        { CKA_KEY_TYPE : CKK_AES        }
-                    ]
-
-                    mech = [
-                        { CKM_AES_CBC : "" },
-                    ]
-
-                # Add keylabel for ALL objects if set
-                if keylabel is not None:
-                    attrs.append({CKA_LABEL : keylabel})
-
-                # Store to database
-                id = db.addtertiary(sobj['id'], tertiarypriv, tertiarypub, encobjauth, attrs, mech)
-
-                # if the keylabel is not set, use the tertiary object id as the keylabel
-                # Normally we would use a transaction to make this atomic, but Pythons
-                # sqlite3 transaction handling is quite odd. So when the keylabel is None, just insert
-                # into the db without that attribute, retrieve the primary key, and then issue an
-                # update. A possible race exists if someone is looking for the key by label between
-                # these operations.
-                # See:
-                #   - https://stackoverflow.com/questions/107005/predict-next-auto-inserted-row-id-sqlite
-                if keylabel is None:
-                    keylabel = str(id)
-                    attrs.append({CKA_LABEL : keylabel})
-                    db.updatetertiaryattrs(id, attrs)
-
-                db.commit()
-
-                print('Added key as label: "{keylabel}"'.format(keylabel=keylabel))
+    def __call__(self, args):
+        super(self.__class__, self).__call__(args)
+        keylabel= args['key_label']
+        print('Added key as label: "{keylabel}"'.format(keylabel=keylabel))
```

Comment:
keylabel is optional and defaults to rowid.

Reply:
Sorry, I dnd't get what changes you are looking for? Do you want me to make the keylabel default to rowid?

Reply:
Fixed it to display the rowid/id if the actual key-label is not provided. by the way, the local variable id was used twice to had to change the 2nd one to rowid